### PR TITLE
Recipe FacesManagedBeansRemoved - Jakarta EE 10

### PR DESCRIPTION
**Pull request by ranuradh**
## What's changed?
image

## Anyone you would like to review specifically?
@timtebeek 

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've added the license header to any new files through `./gradlew licenseFormat`
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 1]
```diff
@@ -254,3 +256,34 @@ recipeList:
       oldFullyQualifiedTypeName: javax.faces.el.ReferenceSyntaxException
       newFullyQualifiedTypeName: jakarta.el.ELException
       ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.jakarta.FacesManagedBeansRemoved
+displayName: TBD
+description: >
+  TBD
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:264]: Could you fill out these values still?

Comment (ranuradh) [reply src/main/resources/META-INF/rewrite/jakarta-ee-10.yml]: sorry done in the next commit

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 2]
```diff
@@ -254,3 +256,34 @@ recipeList:
       oldFullyQualifiedTypeName: javax.faces.el.ReferenceSyntaxException
       newFullyQualifiedTypeName: jakarta.el.ELException
       ignoreDefinition: true
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.jakarta.FacesManagedBeansRemoved
+displayName: Substitute deprecated Faces Managed Beans
+description: >
+  This recipe substitutes Faces Managed Beans, which were deprecated in JavaServer Faces 2.3 and  have been removed from Jakarta Faces 4.0.
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:264]:
```suggestion
description: >
  This recipe substitutes Faces Managed Beans, which were deprecated in JavaServer Faces 2.3 and have been removed from Jakarta Faces 4.0.
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            .activateRecipes("org.openrewrite.java.migrate.jakarta.FacesManagedBeansRemoved"));
+    }
+
+    @Test
+    void updateFacesManagedBean() {
+        rewriteRun(
+          //language=java
+          java("""
+            package com.test;
+
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:44]:
```suggestion
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            @SessionScoped
+            @ManagedProperty
+            @NoneScoped
+            @ViewScoped
+            public class ApplicationBean2 {
+             
+            }              
+            """, """
+            package com.test;
+
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:63]:
```suggestion
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+             
+            @Documented
+            public @interface MyCustomAnnotation{
+                       int studentAge() default 18;            
+            }        
+            """
+          )
+        );
+    }
+
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:102]: I don't think we need a test here that's completely unrelated, given that we reuse existing Yaml recipes here.
```suggestion
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          .recipe(Environment.builder().scanRuntimeClasspath("org.openrewrite.java.migrate.jakarta")
+            .build()
+            .activateRecipes("org.openrewrite.java.migrate.jakarta.FacesManagedBeansRemoved"));
+    }
+
+    @Test
+    void updateFacesManagedBean() {
+        rewriteRun(
+          //language=java
+          java("""
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:42]:
```suggestion
          java(
            """
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            @ApplicationScoped
+            @RequestScoped
+            @SessionScoped
+            @ManagedProperty
+            @NoneScoped
+            @ViewScoped
+            public class ApplicationBean2 {
+             
+            }              
+            """, """
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:59]:
```suggestion
            """,
            """
```

File: src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            import javax.faces.bean.ViewScoped;
+             
+            @ApplicationScoped
+            @RequestScoped
+            @SessionScoped
+            @ManagedProperty
+            @NoneScoped
+            @ViewScoped
+            public class ApplicationBean2 {
+
```
Comment (timtebeek) [inline src/test/java/org/openrewrite/java/migrate/jakarta/FacesManagedBeansRemovedTest.java:57]:
```suggestion
```

File: src/main/resources/META-INF/rewrite/jakarta-ee-10.yml [hunk 9]
```diff
@@ -32,6 +32,8 @@ recipeList:
   - org.openrewrite.java.migrate.jakarta.WsWsocServerContainerDeprecation
   - org.openrewrite.java.migrate.jakarta.ServletCookieBehaviorChangeRFC6265
   - org.openrewrite.java.migrate.BeanDiscovery
+  - org.openrewrite.java.migrate.FacesManagedBeansRemoved
+  - org.openrewrite.java.migrate.jakarta.JavaxMigrationToJakarta
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-ee-10.yml:36]: We already have this one on line 25; don't think we need to run this twice.
```suggestion
```

**Comment by timtebeek** [comment]
Thanks again!

**Pull request merged**